element-connector/src/main/java/org/eclipse/californium/elements/CorrelationStrategy.java

```diff
@@ -0,0 +1,54 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Bosch Software Innovations GmbH and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Bosch Software Innovations GmbH - add flexible correlation context matching
+ *                                      (fix GitHub issue #104)
+ ******************************************************************************/
+package org.eclipse.californium.elements;
+
+/**
+ * Interface for correlation context processing. Enable implementor to flexible
+ * decide on context correlation information.
+ */
+public interface CorrelationStrategy {
+
+	/**
+	 * Return strategy name. Used for logging.
+	 * 
+	 * @return name of strategy.
+	 */
+	String getName();
+
+	/**
+	 * Check, if data should be send.
+	 * 
+	 * @param dataContext correlation context of data to send. May be null, if
+	 *            no special correlation context is required for this data.
+	 * @param connectorContext current correlation context of connector. May be
+	 *            null, if not available or connection is not established.
+	 * @return true, if data should be sent by the connector, false, if data
+	 *         should be suppressed.
+	 */
+	boolean send(CorrelationContext dataContext, CorrelationContext connectorContext);
```

Comment:
"send" is for the followup PR.
It's intended to check the correlation context just before "sending/encrypting" the message.
Currently both implementations are doing the "same checks", but I would like to allow different checkes for matching (mainly coap client side) and sending (mainly coap server side)